addons/l10n_eu_service/models/res_company.py

```diff
@@ -0,0 +1,83 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, fields, models, _
+from .eu_tax_map import EU_TAX_MAP
+
+
+class Company(models.Model):
+    _inherit = 'res.company'
+
+    @api.model
+    def map_all_company_taxes(self):
+        if not self.env['ir.config_parameter'].sudo().get_param('l10n_eu_service.taxes_mapped', False):
+            eu_countries = self.env.ref('base.europe').country_ids
+            companies = self.search([]).filtered(lambda c: c.country_id in eu_countries)
+            companies.map_eu_taxes()
+            self.env['ir.config_parameter'].sudo().set_param('l10n_eu_service.taxes_mapped', True)
+
+    def map_eu_taxes(self):
+        eu_countries = self.env.ref('base.europe').country_ids
+        tax_groups = {}
+        for company in self:
+            invoice_repartition_lines, refund_repartition_lines = company._get_repartition_lines_oss()
+            taxes = self.env['account.tax'].search([('type_tax_use', '=', 'sale'),
```

Comment:
```suggestion
            taxes = self.env['account.tax'].search([
                ('type_tax_use', '=', 'sale'),),
                ('amount_type', '=', 'percent'),
                ('company_id','=', company.id),
                ('tax_group_id', 'not ilike', 'OSS')
            ])
```

Would be more readable.

Reply:
double ), isn'it ?

Reply:
Indeed, but you got my point :p